components/substitutions/index.rst

```diff
@@ -0,0 +1,181 @@
+Substitutions
+=============
+
+.. seo::
+    :description: How to use substitutions in ESPHome
+    :image: settings.svg
+
+ESPHome has a powerful way to reduce repetition in configuration files: substitutions.
+With substitutions, you can have a single generic source file for all nodes of one kind and
+substitute expressions in as required.
+
+.. code-block:: yaml
+
+    substitutions:
+      devicename: livingroom
+      upper_devicename: Livingroom
+
+    esphome:
+      name: $devicename
+      # ...
+
+    sensor:
+    - platform: dht
+      # ...
+      temperature:
+        name: ${upper_devicename} Temperature
+      humidity:
+        name: ${upper_devicename} Humidity
+
+In the top-level ``substitutions`` section, you can put as many key-value pairs as you want. Before
+validating your configuration, ESPHome will automatically replace all occurrences of substitutions
+by their value. The syntax for a substitution is based on bash and is case-sensitive: ``$substitution_key`` or
+``${substitution_key}`` (same).
+
+Two substitution passes are performed allowing compound replacements.
+
+.. code-block:: yaml
+
+    substitutions:
+      foo: yellow
+      bar_yellow_value: !secret yellow_secret
+      bar_green_value: !secret green_secret
+
+    something:
+      test: ${bar_${foo}_value}
+
+.. _YAML-insertion-operator:
+
+YAML insertion operator
```

Comment:
Not sure this fits in the `substitutions` docs.

Reply:
Agreed -- any thoughts on somewhere it _could_ go? I'm drawing a blank. But it is indeed useful...